packages/grid/x-data-grid/src/hooks/features/rows/gridVisibleRowsSelector.ts

```diff
@@ -0,0 +1,17 @@
+import { createSelector } from '../../../utils/createSelector';
+import type { GridStateCommunity } from '../../../models/gridStateCommunity';
+
+/**
+ * @category Visible rows
+ * @ignore - do not document.
+ */
+export const gridVisibleRowsStateSelector = (state: GridStateCommunity) => state.visibleRows;
+
+/**
+ * @category Visible rows
+ * @ignore - do not document.
+ */
+export const gridVisibleRowsLookupSelector = createSelector(
+  gridVisibleRowsStateSelector,
+  (visibleRowsState) => visibleRowsState.lookup,
+);
```

Comment:
In this case, wouldn't it be more simple/lightweight to just use function composition rather than `createSelector`?

Reply:
I guess we can skip caching here, but we'll have to maintain the same signature because we export it.
Is your concern here related to performance?

Reply:
I see. Yeah both performance & complexity, I've been debugging a lot this week and stepping through those selectors is always a bit painful.